Shared/Realm.Shared/RealmConfiguration.cs

```diff
@@ -239,6 +235,56 @@ public bool Equals(RealmConfiguration rhs)
         /// hash table.</returns>
         public override int GetHashCode() => DatabasePath.GetHashCode();
 
+        internal virtual Realm CreateRealm(RealmSchema schema)
+        {
+            var srHandle = new SharedRealmHandle();
+
+            var configuration = new Native.Configuration
+            {
+                Path = DatabasePath,
+                read_only = ReadOnly,
+                delete_if_migration_needed = ShouldDeleteIfMigrationNeeded,
+                schema_version = SchemaVersion
+            };
+
+            Migration migration = null;
+            if (MigrationCallback != null)
+            {
+                migration = new Migration(this, schema);
+                migration.PopulateConfiguration(ref configuration);
+            }
+
+            var srPtr = IntPtr.Zero;
+            try
+            {
+                srPtr = srHandle.Open(configuration, schema, EncryptionKey);
+            }
+            catch (ManagedExceptionDuringMigrationException)
+            {
+                throw new AggregateException("Exception occurred in a Realm migration callback. See inner exception for more details.", migration?.MigrationException);
+            }
+
+            srHandle.SetHandle(srPtr);
+            return new Realm(srHandle, this, schema);
+        }
+
+        internal virtual void DeleteRealm()
+        {
+            // TODO add cache checking when implemented, https://github.com/realm/realm-dotnet/issues/308
+            // when cache checking, uncomment in IntegrationTests.cs RealmInstanceTests.DeleteRealmFailsIfOpenSameThread and add a variant to test open on different thread
+            var lockOnWhileDeleting = new object();
```

Comment:
Sigh, this was just moved here from somewhere else. I'll just undo the refactoring of the deletion logic.